src/test/java/com/myworkspace/reportapp/entity/customer/CustomerTest.java

```diff
@@ -0,0 +1,49 @@
+package com.myworkspace.reportapp.entity.customer;
+
+import com.myworkspace.reportapp.entity.EntityTest;
+import com.myworkspace.reportapp.entity.device.Device;
+import com.myworkspace.reportapp.entity.device.DeviceBase;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CustomerTest extends EntityTest {
```

Comment:
Consider creating utility methods to create test data instances like `Customer`, `DeviceBase`, and `Device`. This can make your test code more concise and reduce duplication.

Reply:
Since you're using `em.flush()` and `em.clear()` after persisting and updating entities, consider whether they are necessary for your test cases. If your tests don't have any dependencies on the persistence context state between tests, these might not be needed.

Reply:
Please try to parametrize those test methods.

Reply:
Prepare csv file with user data to create objects